jobs/cache_maintenance/src/cache_maintenance/metrics.py

```diff
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+import logging
+
+from libcommon.metrics import CustomMetric
+from libcommon.processing_graph import ProcessingStep
+from libcommon.queue import Queue
+from libcommon.simple_cache import get_responses_count_by_kind_status_and_error_code
+
+
+def collect_metrics(processing_steps: list[ProcessingStep]) -> None:
+    logging.info("collecting jobs metrics")
+    queue = Queue()
+    CustomMetric.objects(metric="queue_jobs_total").delete()
```

Comment:
it's a detail, but for a short moment (if the following computations are fast), the metrics will not exist in the DB, which might lead to missing values when prometheus is fetching the metrics.
We could avoid this, but the code would be a bit more complex:
- we could use [upsert_one](http://docs.mongoengine.org/apireference.html?highlight=upsert#mongoengine.queryset.QuerySet.upsert_one) instead of save
- we still want to delete obsolete values, if any. So, we would have to compare the two lists, to upsert the valid metrics, and delete the invalid metrics